### PR TITLE
Add require number of peaks below maximum

### PR DESCRIPTION
**Pull request by niekdejonge**

### CODE REVIEW

File: matchms/filtering/peak_processing/require_number_of_peaks_below_maximum.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import logging
+from typing import Optional
+from matchms.Spectrum import Spectrum
+
+
+logger = logging.getLogger("matchms")
+
+
+def require_number_of_peaks_below_maximum(spectrum_in: Spectrum,
```
Comment (florian-huber) [inline matchms/filtering/peak_processing/require_number_of_peaks_below_maximum.py]:
```suggestion
def require_upper_limit_of_number_of_peaks(spectrum_in: Spectrum,
```

Comment (florian-huber) [reply matchms/filtering/peak_processing/require_number_of_peaks_below_maximum.py]: Also not 100% happy with my suggestion. But I first misunderstood your function name and thought you meant only peaks below the max intensity or something like this...

**Review by florian-huber — APPROVED** [review]
Looks good @niekdejonge . I was just not so sure about the filter name.

Maybe change to
- `require_maximum_number_of_peaks()` ?
(or `require_upper_limit_of_number_of_peaks()` although that feels too long ...)

File: matchms/filtering/peak_processing/require_number_of_peaks_below_maximum.py (hunk 1 above)
Comment (niekdejonge) [reply matchms/filtering/peak_processing/require_number_of_peaks_below_maximum.py]: @florian-huber How about require_less_than_maximum_number_of_peaks?

**Pull request merged**